test/shork_test.gleam

```diff
@@ -105,3 +101,72 @@ pub fn no_table_test() {
     res,
   )
 }
+
+pub fn transaction_test() {
+  let connection = start_default()
+  let insert = fn(connection, name) {
+    let sql =
+      "
+      insert into shorks (name, age, length, species)
+      values (?, 54, 150, 'blue shark')
+      "
+
+    let assert Ok(res) =
+      shork.query(sql)
+      |> shork.parameter(shork.text(name))
+      |> shork.returning({
+        use x0 <- decode.field(0, decode.int)
+        use x1 <- decode.field(1, decode.int)
+        use x2 <- decode.field(2, decode.int)
+
+        decode.success(#(x0, x1, x2))
+      })
+      |> shork.execute(connection)
+    res.rows
+  }
+
+  let assert Ok(#(id1, id2)) = shork.transaction(connection, fn(connection){
+    let res1 = insert(connection, "Kleinrich")
+    let res2 = insert(connection, "Kleinelore")
+
+    let assert Ok(res1) = list.first(res1)
+    let assert Ok(res2) = list.first(res2)
+    Ok(#(res1.0, res2.0))
+  })
+
+  io.debug(id1)
+  io.debug(id2)
```

Comment:
Asserts would be better here.

Reply:
That's true. I need to rework the tests. There are kinda a mess.

Reply:
fixed